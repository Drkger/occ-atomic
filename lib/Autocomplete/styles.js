import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
