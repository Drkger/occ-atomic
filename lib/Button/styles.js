import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};