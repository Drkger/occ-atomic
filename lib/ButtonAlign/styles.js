export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};