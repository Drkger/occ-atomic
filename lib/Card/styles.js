import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;