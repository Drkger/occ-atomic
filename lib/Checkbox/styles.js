import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
