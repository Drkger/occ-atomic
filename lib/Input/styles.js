import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    container: props => props.theme != 'flat' ? {
        marginBottom:'16px',
        paddingBottom:'11px',
        position:'relative'
    } : {
        position:'relative'
    },
    default: props => props.theme != 'flat' && {
        color:colors.grey4,
        '& $input': {
            borderColor:colors.grey1,
        },
        '& $placeholder': {
            top:'20px',
            fontSize:'16px',
            color:colors.grey6
        }
    },
    disabled: props => props.theme != 'flat' && {
        color:colors.grey1,
        '& $input': {
            borderColor:colors.grey1,
            color:colors.grey1
        },
        '& $placeholder': {
            top:'20px',
            fontSize:'16px',
            color:colors.grey2
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    focus: props => props.theme != 'flat' && {
        color:colors.blue,
        '& $inputWrap:after': {
            width:'100%',
            left:'0'
        },
        '& $placeholder': {
            color:colors.blue
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    complete: props => props.theme != 'flat' && {
        color:colors.black2,
        '& $input': {
            borderColor:colors.black2,
        },
        '& $placeholder': {
            color:colors.black2
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    correct: props => props.theme != 'flat' && {
        color:colors.green,
        '& $input': {
            borderColor:colors.green,
        },
        '& $placeholder': {
            color:colors.green
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    error: props => props.theme != 'flat' && {
        color:colors.red,
        '& $input': {
            borderColor:colors.red,
        },
        '& $placeholder': {
            color:colors.red
        },
        '& $pushHolder': {
            left:'0'
        }
    },
    label: props => props.theme != 'flat' ? {
        boxSizing:'border-box',
        float:'left',
        display:'inline-block',
        maxWidth:'100%',
        fontFamily:fonts.body,
        fontSize:'11px',
        fontWeight:'normal',
        marginBottom:'0',
        transition:'0.3s all'
    } : {
        fontFamily:fonts.body,
        fontSize:'14px',
        lineHeight:'24px',
        color:colors.white
    },
    left: props => props.theme != 'flat' && {
        float:'left'
    },
    push: props => props.theme != 'flat' && {
        paddingTop:'15px'
    },
    counter: props => props.theme != 'flat' && {
        position:'absolute',
        top:'0',
        right:'0'
    },
    placeholder: props => props.theme != 'flat' && {
        position:'absolute',
        pointerEvents:'none',
        fontSize:'11px',
        top:'0'
    },
    pushHolder: props => props.theme != 'flat' && {
        left:'24px'
    },
    inputWrap: props => props.theme != 'flat' && {
        boxSizing:'border-box',
        position:'relative',
        '&:after': {
            content:'""',
            display:'block',
            position:'absolute',
            background:colors.blue,
            transition:'0.3s all',
            bottom:'0',
            left:'50%',
            width:'0',
            height:'1px'
        },
    },
    input: props => props.theme != 'flat' ? {
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        width:'100%',
        height:'34px',
        color:colors.black,
        fontFamily:fonts.body,
        fontWeight:'normal',
        fontSize:'16px',
        lineHeight:'16px',
        background:'transparent',
        border:'0',
        borderBottom:`1px solid ${colors.grey1}`,
        padding:'10px 0',
        marginBottom:'4px',
        outline:'none',
        transition:'0.3s all',
        '&::placeholder': {
            color:colors.grey6
        }
    } : {
        boxSizing:'border-box',
        display:'block',
        width:'100%',
        height:'48px',
        paddingLeft:'10px',
        paddingRight:'10px',
        background:colors.white,
        color:colors.black,
        fontFamily:fonts.body,
        fontWeight:'normal',
        fontSize:'14px',
        lineHeight:'24px',
        border:'0',
        borderRadius:'3px',
        outline:'none',
        '&::placeholder': {
            color:colors.grey7
        }
    },
    textarea: props => props.theme != 'flat' && {
        padding:'5px 0',
        lineHeight:'22px'
    },
    select: props => props.theme != 'flat' && {
        padding:'0',
        appearance:'none',
        background:icons.base(icons.arrowDown.icon(['#4E6273'])),
        backgroundSize:'24px 24px',
        backgroundPosition:'calc(100% - 5px) 5px'
    },
    expanded: props => props.theme != 'flat' && {
        height:'80px'
    },
    hasIcon: props => props.theme != 'flat' && {
        paddingLeft:'24px'
    },
    icon: props => props.theme != 'flat' && {
        position:'absolute',
        left:'0',
        top:'19px'
    },
    hasClear: props => props.theme != 'flat' && {
        paddingRight:'24px'
    },
    clear: {
        background:'none',
        border:'0',
        padding:'0',
        margin:'0',
        position:'absolute',
        right:props => props.theme != 'flat' ? '0' : '10px',
        top:props => props.theme != 'flat' ? '20px' : '35px',
        outline:'0',
        cursor:'pointer'
    }
};