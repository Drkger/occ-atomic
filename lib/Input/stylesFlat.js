import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';

export default {
    container: {
        position:'relative'
    },
    default: {},
    disabled: {},
    focus: {},
    complete: {},
    correct: {},
    error: {},
    label: {
        fontFamily:fonts.body,
        fontSize:'14px',
        lineHeight:'24px',
        color:colors.white
    },
    left: {},
    push: {},
    counter: {},
    placeholder: {},
    pushHolder: {},
    inputWrap: {},
    input: {
        boxSizing:'border-box',
        display:'block',
        width:'100%',
        height:'48px',
        paddingLeft:'10px',
        paddingRight:'10px',
        background:colors.white,
        color:colors.black,
        fontFamily:fonts.body,
        fontWeight:'normal',
        fontSize:'14px',
        lineHeight:'24px',
        border:'0',
        borderRadius:'3px',
        outline:'none',
        '&::placeholder': {
            color:colors.grey7
        }
    },
    textarea: {},
    select: {},
    expanded: {},
    hasIcon: {},
    icon: {},
    hasClear: {},
    clear: {
        background:'none',
        border:'0',
        padding:'0',
        margin:'0',
        position:'absolute',
        right:'10px',
        top:'35px',
        outline:'0',
        cursor:'pointer'
    }
};