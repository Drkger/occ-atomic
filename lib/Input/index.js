import injectSheet from 'react-jss';
import Input from './Input';
import styles from './styles';

export default injectSheet(styles)(Input);