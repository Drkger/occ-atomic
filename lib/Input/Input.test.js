import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';
import styles from './styles';
import InputJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Input", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the right html tag', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper.find('.input').type()).toBe('input');
        wrapper.setProps({type:'textarea'});
        expect(wrapper.find('.input').type()).toBe('textarea');
        wrapper.setProps({type:'select', options:[]});
        expect(wrapper.find('.input').type()).toBe('select');
    });

    it('renders the right type of input', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper.find('.input').prop('type')).toBe('text');
        wrapper.setProps({type:'email'});
        expect(wrapper.find('.input').prop('type')).toBe('email');
        wrapper.setProps({type:'number'});
        expect(wrapper.find('.input').prop('type')).toBe('number');
    });

    it('sets the maxLength property', () => {
        const wrapper = shallow(
            <Input classes={classes} maxLength={2} />
        );
        expect(wrapper.find('.input').prop('maxLength')).toBe(2);
    });

    it('shows a counter with the remaining characters', () => {
        const wrapper = shallow(
            <Input classes={classes} maxLength={50} counter />
        );
        expect(wrapper.find('.counter').length).toBe(0);
        wrapper.setState({status:'focus'});
        expect(wrapper.find('.counter').length).toBe(1);
        expect(wrapper.find('.counter').text()).toBe('50');
        wrapper.setState({value:'1234567890'});
        expect(wrapper.find('.counter').text()).toBe('40');
    });

    it('renders a label and a placeholder in the input', () => {
        const wrapper = shallow(
            <Input classes={classes} label="Label" placeholder="Placeholder" />
        );
        expect(wrapper.find('.label').length).toBe(1);
        expect(wrapper.find('.label').text()).toBe('Label');
        expect(wrapper.find('.input').prop('placeholder')).toBe('Placeholder');
    });

    it('renders a placeholder that works as a label on focus', () => {
        const wrapper = shallow(
            <Input classes={classes} placeholder="Placeholder" />
        );
        expect(wrapper.find('.placeholder').length).toBe(1);
        expect(wrapper.find('.placeholder').text()).toBe('Placeholder');
        expect(wrapper.find('.container').hasClass('push')).toBe(true);
    });

    it('renders a static field when disabled', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper.find('.container').hasClass('disabled')).toBe(false);
        expect(wrapper.find('.input').type()).toBe('input');
        wrapper.setProps({disabled: true});
        expect(wrapper.find('.container').hasClass('disabled')).toBe(true);
        expect(wrapper.find('.input').type()).toBe('label');
    });

    it('sets the autoFocus property', () => {
        const wrapper = shallow(
            <Input classes={classes} autoFocus />
        );
        expect(wrapper.find('.input').prop('autoFocus')).toBe(true);
    });

    it('shows an icon that clears the value on click', () => {
        const wrapper = shallow(
            <Input classes={classes} clear />
        );
        expect(wrapper.find('.clear').length).toBe(0);
        wrapper.setState({value:'Value'});
        expect(wrapper.find('.clear').length).toBe(1);
        wrapper.find('.clear').simulate('click');
        expect(wrapper.find('.clear').length).toBe(0);
        expect(wrapper.state('value')).toBe('');
    });

    it('shows an error message', () => {
        const wrapper = shallow(
            <Input classes={classes} error="This field has an error" />
        );
        expect(wrapper.find('.error').length).toBe(0);
        wrapper.setState({touched: true});
        expect(wrapper.find('.error').length).toBe(2);
        expect(wrapper.find('.error').at(0).hasClass('container')).toBe(true);
        expect(wrapper.find('.error').at(1).text()).toBe('This field has an error');
    });

    it('shows the field as a valid value', () => {
        const wrapper = shallow(
            <Input classes={classes} valid />
        );
        expect(wrapper.find('.container').hasClass('correct')).toBe(false);
        wrapper.setState({touched: true});
        expect(wrapper.find('.container').hasClass('correct')).toBe(true);
    });

    it('sets a default value', () => {
        const wrapper = shallow(
            <Input classes={classes} valueProp="Value" />
        );
        expect(wrapper.state('value')).toBe('Value');
    });

    it('calls the onFocus function', () => {
        const onFocus = jest.fn();
        const wrapper = shallow(
            <Input classes={classes} onFocus={onFocus} />
        );
        wrapper.find('.input').simulate('focus', {target: {}});
        expect(onFocus.mock.calls.length).toBe(1);
    });

    it('calls the onBlur function', () => {
        const onBlur = jest.fn();
        const wrapper = shallow(
            <Input classes={classes} onBlur={onBlur} />
        );
        wrapper.find('.input').simulate('blur', {target: {value: 'Value'}});
        expect(onBlur.mock.calls.length).toBe(1);
    });

    it('calls the onChange function', () => {
        const onChange = jest.fn();
        const wrapper = shallow(
            <Input classes={classes} onChange={onChange} />
        );
        wrapper.find('.input').simulate('change', {target: {}});
        expect(onChange.mock.calls.length).toBe(1);
    });

    it('calls the onClear function', () => {
        const onClear = jest.fn();
        const wrapper = shallow(
            <Input classes={classes} clear onClear={onClear} valueProp="Value" />
        );
        wrapper.find('.clear').simulate('click');
        expect(onClear.mock.calls.length).toBe(1);
    });

    it('calls the onRef function', () => {
        const onRef = jest.fn();
        const wrapper = shallow(
            <Input classes={classes} onRef={onRef} />
        );
        wrapper.update();
        expect(onRef.mock.calls.length).toBe(1);
    });

    it('renders the options of the select', () => {
        const options = [
            {
                value: 1,
                text: 'Option 1'
            },
            {
                value: 2,
                text: 'Option 2'
            }
        ];
        const wrapper = shallow(
            <Input classes={classes} type="select" options={options} />
        );
        expect(wrapper.find('option').length).toBe(2);
        expect(wrapper.find('option').at(0).text()).toBe('Option 1');
        expect(wrapper.find('option').at(0).prop('value')).toBe(1);
    });

    it('shows an icon', () => {
        const wrapper = shallow(
            <Input classes={classes} />
        );
        expect(wrapper.find('.icon').length).toBe(0);
        wrapper.setProps({iconName: 'search'});
        expect(wrapper.find('.icon').length).toBe(1);
    });

});

describe("InputJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <InputJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Input styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});