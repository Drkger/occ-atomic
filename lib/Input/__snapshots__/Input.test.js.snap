// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    autoFocus={false}
    classes={
      Object {
        "clear": "clear",
        "complete": "complete",
        "container": "container",
        "correct": "correct",
        "counter": "counter",
        "default": "default",
        "disabled": "disabled",
        "error": "error",
        "expanded": "expanded",
        "focus": "focus",
        "hasClear": "hasClear",
        "hasIcon": "hasIcon",
        "icon": "icon",
        "input": "input",
        "inputWrap": "inputWrap",
        "label": "label",
        "left": "left",
        "placeholder": "placeholder",
        "push": "push",
        "pushHolder": "pushHolder",
        "select": "select",
        "textarea": "textarea",
      }
    }
    clear={false}
    counter={false}
    disabled={false}
    maxLength={100}
    selectOnFocus={false}
    type="text"
    valueProp=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        undefined,
        false,
        <div
          className="inputWrap"
        >
          <input
            autoFocus={false}
            className="input"
            id={undefined}
            maxLength={100}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>,
        "",
        undefined,
      ],
      "className": "container default",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      undefined,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            autoFocus={false}
            className="input"
            id={undefined}
            maxLength={100}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            placeholder=""
            type="text"
            value=""
          />,
          "className": "inputWrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": false,
            "className": "input",
            "id": undefined,
            "maxLength": 100,
            "name": undefined,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onKeyUp": [Function],
            "placeholder": "",
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      "",
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          undefined,
          false,
          <div
            className="inputWrap"
          >
            <input
              autoFocus={false}
              className="input"
              id={undefined}
              maxLength={100}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyUp={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </div>,
          "",
          undefined,
        ],
        "className": "container default",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        undefined,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              autoFocus={false}
              className="input"
              id={undefined}
              maxLength={100}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyUp={[Function]}
              placeholder=""
              type="text"
              value=""
            />,
            "className": "inputWrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": false,
              "className": "input",
              "id": undefined,
              "maxLength": 100,
              "name": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onKeyUp": [Function],
              "placeholder": "",
              "type": "text",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        "",
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Input styles matches the snapshot 1`] = `
Object {
  "clear": Object {
    "background": "none",
    "border": "0",
    "cursor": "pointer",
    "margin": "0",
    "outline": "0",
    "padding": "0",
    "position": "absolute",
    "right": [Function],
    "top": [Function],
  },
  "complete": [Function],
  "container": [Function],
  "correct": [Function],
  "counter": [Function],
  "default": [Function],
  "disabled": [Function],
  "error": [Function],
  "expanded": [Function],
  "focus": [Function],
  "hasClear": [Function],
  "hasIcon": [Function],
  "icon": [Function],
  "input": [Function],
  "inputWrap": [Function],
  "label": [Function],
  "left": [Function],
  "placeholder": [Function],
  "push": [Function],
  "pushHolder": [Function],
  "select": [Function],
  "textarea": [Function],
}
`;

exports[`InputJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Input)
    autoFocus={false}
    clear={false}
    counter={false}
    disabled={false}
    maxLength={100}
    selectOnFocus={false}
    type="text"
    valueProp=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoFocus": false,
      "classes": Object {
        "clear": "Input-clear-0-1-44 Input-clear-0-1-22",
        "complete": "Input-complete-0-1-27 Input-complete-0-1-5",
        "container": "Input-container-0-1-23 Input-container-0-1-1",
        "correct": "Input-correct-0-1-28 Input-correct-0-1-6",
        "counter": "Input-counter-0-1-33 Input-counter-0-1-11",
        "default": "Input-default-0-1-24 Input-default-0-1-2",
        "disabled": "Input-disabled-0-1-25 Input-disabled-0-1-3",
        "error": "Input-error-0-1-29 Input-error-0-1-7",
        "expanded": "Input-expanded-0-1-40 Input-expanded-0-1-18",
        "focus": "Input-focus-0-1-26 Input-focus-0-1-4",
        "hasClear": "Input-hasClear-0-1-43 Input-hasClear-0-1-21",
        "hasIcon": "Input-hasIcon-0-1-41 Input-hasIcon-0-1-19",
        "icon": "Input-icon-0-1-42 Input-icon-0-1-20",
        "input": "Input-input-0-1-37 Input-input-0-1-15",
        "inputWrap": "Input-inputWrap-0-1-36 Input-inputWrap-0-1-14",
        "label": "Input-label-0-1-30 Input-label-0-1-8",
        "left": "Input-left-0-1-31 Input-left-0-1-9",
        "placeholder": "Input-placeholder-0-1-34 Input-placeholder-0-1-12",
        "push": "Input-push-0-1-32 Input-push-0-1-10",
        "pushHolder": "Input-pushHolder-0-1-35 Input-pushHolder-0-1-13",
        "select": "Input-select-0-1-39 Input-select-0-1-17",
        "textarea": "Input-textarea-0-1-38 Input-textarea-0-1-16",
      },
      "clear": false,
      "counter": false,
      "disabled": false,
      "maxLength": 100,
      "selectOnFocus": false,
      "type": "text",
      "valueProp": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": false,
        "classes": Object {
          "clear": "Input-clear-0-1-44 Input-clear-0-1-22",
          "complete": "Input-complete-0-1-27 Input-complete-0-1-5",
          "container": "Input-container-0-1-23 Input-container-0-1-1",
          "correct": "Input-correct-0-1-28 Input-correct-0-1-6",
          "counter": "Input-counter-0-1-33 Input-counter-0-1-11",
          "default": "Input-default-0-1-24 Input-default-0-1-2",
          "disabled": "Input-disabled-0-1-25 Input-disabled-0-1-3",
          "error": "Input-error-0-1-29 Input-error-0-1-7",
          "expanded": "Input-expanded-0-1-40 Input-expanded-0-1-18",
          "focus": "Input-focus-0-1-26 Input-focus-0-1-4",
          "hasClear": "Input-hasClear-0-1-43 Input-hasClear-0-1-21",
          "hasIcon": "Input-hasIcon-0-1-41 Input-hasIcon-0-1-19",
          "icon": "Input-icon-0-1-42 Input-icon-0-1-20",
          "input": "Input-input-0-1-37 Input-input-0-1-15",
          "inputWrap": "Input-inputWrap-0-1-36 Input-inputWrap-0-1-14",
          "label": "Input-label-0-1-30 Input-label-0-1-8",
          "left": "Input-left-0-1-31 Input-left-0-1-9",
          "placeholder": "Input-placeholder-0-1-34 Input-placeholder-0-1-12",
          "push": "Input-push-0-1-32 Input-push-0-1-10",
          "pushHolder": "Input-pushHolder-0-1-35 Input-pushHolder-0-1-13",
          "select": "Input-select-0-1-39 Input-select-0-1-17",
          "textarea": "Input-textarea-0-1-38 Input-textarea-0-1-16",
        },
        "clear": false,
        "counter": false,
        "disabled": false,
        "maxLength": 100,
        "selectOnFocus": false,
        "type": "text",
        "valueProp": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
